# HG changeset patch
# User David Vrabel <david.vrabel@citrix.com>
# Date 1381739043 -7200
# Node ID a446e8455793bb1d4a797d73541400392b585fcb
# Parent  a8e02fdab820f3d3ccffd8887a5c6583d2bc2306
libxc: add xc_domain_set_max_evtchn()

Add xc_domain_set_max_evtchn(), a wrapper around the
DOMCTL_set_max_evtchn hypercall.

Signed-off-by: David Vrabel <david.vrabel@citrix.com>
Acked-by: Ian Campbell <ian.campbell@citrix.com>
Acked-by: Keir Fraser <keir@xen.org>

diff -r a8e02fdab820 -r a446e8455793 tools/libxc/xc_domain.c
--- a/tools/libxc/xc_domain.c	Mon Oct 14 10:23:10 2013 +0200
+++ b/tools/libxc/xc_domain.c	Mon Oct 14 10:24:03 2013 +0200
@@ -1766,6 +1766,17 @@ int xc_domain_set_virq_handler(xc_interf
     return do_domctl(xch, &domctl);
 }
 
+int xc_domain_set_max_evtchn(xc_interface *xch, uint32_t domid,
+                             uint32_t max_port)
+{
+    DECLARE_DOMCTL;
+
+    domctl.cmd = XEN_DOMCTL_set_max_evtchn;
+    domctl.domain = domid;
+    domctl.u.set_max_evtchn.max_port = max_port;
+    return do_domctl(xch, &domctl);
+}
+
 /*
  * Local variables:
  * mode: C
diff -r a8e02fdab820 -r a446e8455793 tools/libxc/xenctrl.h
--- a/tools/libxc/xenctrl.h	Mon Oct 14 10:23:10 2013 +0200
+++ b/tools/libxc/xenctrl.h	Mon Oct 14 10:24:03 2013 +0200
@@ -847,6 +847,18 @@ int xc_domain_set_access_required(xc_int
  */
 int xc_domain_set_virq_handler(xc_interface *xch, uint32_t domid, int virq);
 
+/**
+ * Set the maximum event channel port a domain may bind.
+ *
+ * This does not affect ports that are already bound.
+ *
+ * @param xch a handle to an open hypervisor interface
+ * @param domid the domain id
+ * @param max_port maximum port number
+ */
+int xc_domain_set_max_evtchn(xc_interface *xch, uint32_t domid,
+                             uint32_t max_port);
+
 /*
  * CPUPOOL MANAGEMENT FUNCTIONS
  */
